rmw_connext_cpp/src/rmw_publisher.cpp

```diff
@@ -252,9 +252,7 @@ rmw_create_publisher(
     DDS_String_free(topic_str);
     topic_str = nullptr;
   }
-  if (publisher) {
```

Comment:
This is necessary of the goto in this line is triggered: https://github.com/ros2/rmw_connext/pull/258/files#diff-3bcd912088bec36df10ef09cd4c564b1R220